### PR TITLE
Jms activity class initial clean up

### PR DESCRIPTION
**Pull request by jordy1611**
### What’s this PR do?
* Initial refactor of activity class
* Gets rid of user-repo methods being used in methods that have nothing to do with user-repo
* Some refactoring of testing but still needs a major refactoring
### Where should the reviewer start?
* Activity.js
* activity-test.js
### How should this be manually tested?
* npm test
* Check the website to make sure all is displaying
### Any background context you want to provide?
* The friends section of the class has not been touched
* The refactored and new methods should help quickly update the friends section
* The testing for activity still needs to be updated
* scripts file needs to be updated with edited or new methods

### CODE REVIEW

File: test/activity-test.js [hunk 1]
```diff
@@ -212,10 +212,10 @@ describe('Activity', function() {
```
Comment (jordy1611) [inline test/activity-test.js:212]: This still isn't taking Carly's test data files. Next big merge should fix it

File: src/Activity.js [hunk 2]
```diff
@@ -1,54 +1,126 @@
...
-  //   return parseFloat(((userStepsByDate.numSteps * userRepo.strideLength) / 5280).toFixed(1)); //how is userRepo.strideLength even being accessed?
-  //   // userRepo is array of user objects but how does that have a strideLength property?
-  //   // userRepo is improperly named
-  // }
+  //// begin unused methods
+  getMilesByDate(id, date, user, userRepo) {  //returns miles walked on a specific date
+    // THIS IS NEVER USED needs to be displayed on page
+    let userStepsByDate = this.getUserDataForDay(id, date, 'numSteps').amount;
+    return parseFloat(((userStepsByDate * user.strideLength) / 5280).toFixed(1));
+  }
```
Comment (relyt4me) [inline src/Activity.js:10]: This could be a really cool section on the widgets

File: src/Activity.js [hunk 3]
```diff
@@ -1,54 +1,126 @@
...
+    const weekData = this.getUserWeekData(id, date, 'minutesActive');
+    const weeklyAverage = (weekData.reduce((sum, day) => {
+      return sum += day.amount
+    }, 0) / 7)
+    return parseFloat(weeklyAverage.toFixed(1));
+    // return `{parseFloat(weeklyAverage.toFixed(1))} minutes active`
+    // return parseFloat((userRepo.getWeekFromDate(date, id, this.activityData).reduce((acc, elem) => { //acc and elem are terrible names
+    //   return acc += elem.minutesActive;
+    // }, 0) / 7).toFixed(1));
+  }
```
Comment (relyt4me) [inline src/Activity.js:33]: This is for just one user right? I like the logic for it a lot and might use it in Sleep.

File: src/Activity.js [hunk 4]
```diff
@@ -1,54 +1,126 @@
...
+
+  getUserDataForDay(id, date, dataType) {
+    let todayActivities = this.activityData.find(data => data.date === date && data.userID === id);
+    let todayActivity = {};
+    todayActivity.date = date;
+    todayActivity.amount = todayActivities[dataType];
+    todayActivity.type = (dataType === 'numSteps') ? 'steps' : dataType;
+    return todayActivity
+    // returns { date: '2019/06/15', amount: 3577, type: 'steps' }
```
Comment (relyt4me) [inline src/Activity.js:84]: This format is going to make the dom maniuplation so much easier

File: src/Activity.js (hunk 3 above)
Comment (carlymclift) [reply src/Activity.js:33]: Yeah, get logic here!

**Pull request merged**